### PR TITLE
Implement `karmdactl register` for joining clusters with PULL mode

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <531187475@qq.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Register a new member cluster.

```
[root@master68 lonelyCZ]# ./karmadactl register 10.10.103.67:32443 --cluster-name member68 --token hlvq6f.szq0580jalq1v9o2 --discovery-token-ca-cert-hash sha256:5d9fc2b87c186aaac219f883b4267704a94a0b2e2cc61b2c03bed2699dba9a55  --karmada-agent-image='swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-agent:latest'
[preflight] Running pre-flight checks
[prefligt] All pre-flight checks were passed
[karmada-agent-start] Waiting to perform the TLS Bootstrap
[karmada-agent-start] Waiting to construct karmada-agent kubeconfig
[karmada-agent-start] Waiting the necessary secret and RBAC
[karmada-agent-start] Waiting karmada-agent Deployment
W0823 15:34:47.857253     739 check.go:52] pod: karmada-agent-5d659b4746-fk8gl not ready. status: ContainerCreating
I0823 15:34:48.863019     739 check.go:49] pod: karmada-agent-5d659b4746-fk8gl is ready. status: Running

cluster(member68) is joined successfully
```

On karmada control plane.
```
[root@master67 karmada]# kapi get clusters
NAME       VERSION   MODE   READY   AGE
member68   v1.22.3   Pull   True    36s
```

If `karmada-agent` already installed in the member clusters.

```
[root@master68 lonelyCZ]# ./karmadactl register 10.10.103.67:32443 --cluster-name member68 --token hlvq6f.szq0580jalq1v9o2 --discovery-token-ca-cert-hash sha256:5d9fc2b87c186aaac219f883b4267704a94a0b2e2cc61b2c03bed2699dba9a55  --karmada-agent-image='swr.ap-southeast-1.myhuaweicloud.com/karmada/karmada-agent:latest'
[preflight] Running pre-flight checks
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR]: /etc/karmada/karmada-agent.conf already exists
        [ERROR]: /etc/karmada/pki/ca.crt already exists
        [ERROR]: karmada-system/karmada-kubeconfig Secret already exists
        [ERROR]: karmada-system/karmada-agent-sa ServiceAccount already exists
        [ERROR]: karmada-system/karmada-agent Deployment already exists

[preflight] Please check the above errors
```

**Which issue(s) this PR fixes**:
Part of #2282 

Fixes https://github.com/karmada-io/karmada/issues/1363 https://github.com/karmada-io/karmada/issues/1648

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl: implement `register` command for joining PULL mode cluster
```

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
please assign it to me once it gets ready.

File: pkg/karmadactl/register.go [hunk 1]
```diff
@@ -0,0 +1,981 @@
...
+		if err != nil {
+			return fmt.Errorf("failed to read kubeconfig file %s, err: %w", o.KubeConfig, err)
+		}
+
+		config, err := clientcmd.Load(configBytes)
+		if err != nil {
+			return err
+		}
+
+		o.ClusterName = config.Contexts[config.CurrentContext].Cluster
```
Comment (lonelyCZ) [inline pkg/karmadactl/register.go:266]: If users don't set `--cluster-name`, currently using default clusterName of `kubeconfig` of member cluster. But sometimes, most users install kubernetes in member cluster by default way, lead to the default clusterName is `kubernetes`.

```
[root@master68 lonelyCZ]# cat /root/.kube/config
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: ......
    server: https://10.10.103.68:6443
  name: kubernetes
contexts:
- context:
    cluster: kubernetes
    user: kubernetes-admin
  name: kubernetes-admin@kubernetes
current-context: kubernetes-admin@kubernetes
```

So how do we set default cluaterName if users don't set `--cluster-name`?

Comment (RainbowMango) [reply pkg/karmadactl/register.go:266]: > But sometimes, most users install kubernetes in member cluster by default way,

I don't know much about how users configured their clusters.
But I think if users are using multi-clusters, they should take good care of it. We can assume the default cluster name is well-configured. 
Yes, I mean, keeping the default is fine to me.

Comment (lonelyCZ) [reply pkg/karmadactl/register.go:266]: Agree!

**Comment by lonelyCZ** [comment]
Please retest it @RainbowMango

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**